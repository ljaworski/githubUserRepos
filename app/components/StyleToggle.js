import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Switch } from "native-base";
import { toggleDarkMode } from "../actions/appSettingsActions";

export function StyleToggle({ darkMode, toggleDarkMode }) {
  return <Switch value={darkMode} onValueChange={toggleDarkMode} />;
}

StyleToggle.propTypes = {
  darkMode: PropTypes.bool,
  toggleDarkMode: PropTypes.func
};

const mapStateToProps = ({ appSettings: { darkMode } }) => ({
  darkMode
});

const mapDispatchToProps = dispatch => ({
  toggleDarkMode: () => dispatch(toggleDarkMode())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StyleToggle);
