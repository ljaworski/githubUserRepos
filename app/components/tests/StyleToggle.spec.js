import React from "react";
import { StyleToggle } from "../StyleToggle";
import renderer from "react-test-renderer";

const props = {
  darkMode: false,
  toggleDarkMode: () => {}
};

test("renders correctly", () => {
  const styleToggle = renderer.create(<StyleToggle {...props} />).toJSON();
  expect(styleToggle).toMatchSnapshot();
});
