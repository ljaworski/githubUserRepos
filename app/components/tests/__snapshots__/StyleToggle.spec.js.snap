// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSwitch
  accessibilityRole="button"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Object {
        "marginBottom": -5,
        "marginTop": -5,
      },
    ]
  }
  value={false}
/>
`;
